src/Orleans/GrainDirectory/MultiClusterRegistrationStrategy.cs

```diff
@@ -8,7 +9,7 @@ namespace Orleans.GrainDirectory
     /// Strategy objects are used as keys to select the proper registrar.
     /// </summary>
     [Serializable]
-    internal abstract class MultiClusterRegistrationStrategy
+    public abstract class MultiClusterRegistrationStrategy
```

Comment:
should we have a public interface `IMultiClusterRegistrationStrategy` so that we don't need to make this internal class public? 
if yes I will start to refactor this one.